File: src/main/kotlin/com/indefiniteStudies/model/entities/Tag.kt
package com.indefiniteStudies.model.entities

import org.jetbrains.exposed.sql.Column
import org.jetbrains.exposed.sql.Table

object Tag : Table("tags") {
    val id: Column<Int> = integer("id").autoIncrement()
    val name: Column<String> = varchar("varchar", 256)
    val state: Column<String> = varchar("state", 256)
}

File: internal/db/entities/tag.go
package entities

type Tag struct {
	Id    int
	Name  string
	State string
}

const (
	TAG_STATE_NEW     string = "NEW"
	TAG_STATE_BLOCKED string = "BLOCKED"
	TAG_STATE_DELETED string = "DELETED"
)

func GetPossibleTagStates() []string {
	return []string{TAG_STATE_NEW, TAG_STATE_BLOCKED, TAG_STATE_DELETED}
}
